music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,552 @@
+            self.container.insert(0, meter.TimeSignature('4/4'))  # Placeholder
+
+        self.currentKeyString: str = ''
+        self.prepSequentialListOfLines()
+
+    def _makeContainer(self,
+                       obj: Union[stream.Stream, list]):
+        '''
+        Makes a placeholder container for the unusual cases where this class is called on
+        generic- or non-stream object as opposed to
+        a :class:`~music21.stream.Score`, :class:`~music21.stream.Part`,
+        or :class:`~music21.stream.Measure`.
+        '''
+        m = stream.Measure()
+        for x in obj:
+            m.append(x)
+        self.container = stream.Part()
+        self.container.insert(0, m)
+
+    def prepTitle(self,
+                  md: metadata):
+        '''
+        Attempt to prepare a single work title from the score metadata looking at each of
+        the title, movementNumber and movementName attributes.
+        Failing that, a placeholder 'Title unknown' stands in.
+
+        >>> s = stream.Score()
+        >>> rnScore = romanText.writeRoman.RnWriter(s)
+        >>> rnScore.title
+        'Title unknown'
+
+        >>> s.insert(0, metadata.Metadata())
+        >>> s.metadata.title = 'Fake title'
+        >>> s.metadata.movementNumber = 123456789
+        >>> s.metadata.movementName = 'Fake movementName'
+        >>> rnScoreWithMD = romanText.writeRoman.RnWriter(s)
+        >>> rnScoreWithMD.title
+        'Fake title - No.123456789: Fake movementName'
+        '''
+
+        workingTitle = []
+
+        if md.title:
+            workingTitle.append(md.title)
+        if md.movementNumber:
+            workingTitle.append(f'- No.{md.movementNumber}:')  # Spaces later
+        if md.movementName:
+            if md.movementName != md.title:
+                workingTitle.append(md.movementName)
+
+        if len(workingTitle) > 0:
+            self.title = ' '.join(workingTitle)
+
+# ------------------------------------------------------------------------------
+
+    def prepSequentialListOfLines(self):
+        '''
+        Prepares a sequential list of text lines, with time signatures and Roman numerals
+        adding this to the (already prepared) metadata preamble ready for printing.
+
+        >>> p = stream.Part()
+        >>> m = stream.Measure()
+        >>> m.insert(0, meter.TimeSignature('4/4'))
+        >>> m.insert(0, roman.RomanNumeral('V', 'G'))
+        >>> p.insert(0, m)
+        >>> testCase = romanText.writeRoman.RnWriter(p)
+        >>> testCase.combinedList[-1]  # Last entry, after the metadata
+        'm0 b1 G: V'
+        '''
+
+        for thisMeasure in self.container.getElementsByClass('Measure'):
+            # TimeSignatures
+            tsThisMeasure = thisMeasure.getElementsByClass('TimeSignature')
+            if tsThisMeasure:
+                firstTS = tsThisMeasure[0]
+                self.combinedList.append(f'Time Signature: {firstTS.ratioString}')
```

Comment:
The initial Time Signature should go into the header section.  My sense is that the header and the body should be stored separately and only combined later, so that the blank line between the two is consistent.

Reply:
Thanks for raising this.

Earlier versions had this kind of header vs body separation but (if I remember right) @jacobtylerwalls and I decided to get rid of it as part of streamlining the process and avoid unnecessary storage/duplication (and the potential for confusion and errors that invites).

For instance, the handling of Opus objects (approved in this review) would have to change. Should those cases be stored as Header 1, Body 1, Header 2, Body 2, etc.?

All in all, I'd vote to leave this as it is (one continuous, unseparated list) but am happy to be overruled and implement an alternative.